core/src/main/scala/org/apache/spark/ui/ToolTips.scala

```diff
@@ -62,6 +62,13 @@ private[spark] object ToolTips {
     """Time that the executor spent paused for Java garbage collection while the task was
        running."""
 
+  val PEAK_EXECUTION_MEMORY =
+    """Execution memory refers to the memory used by internal data structures created during
+       shuffles, aggregations and joins when Tungsten is enabled. The value of this accumulator
+       should be approximately the sum of the peak sizes across all such data structures created
+       in this task. For SQL jobs, this only tracks all unsafe operators, broadcast joins, and
+       external sort."""
```

Comment:
> I like the idea of something like spark.execution.memoryFraction to describe the fraction of memory that is reserved for internal use by Spark's own execution operators and not by caching or user-code.

yeah I like that more too ... but especially as you put it that way, I wonder should it have "internal" in there somewhere?  I still worry its going to sound too much like its the total memory usage for the task (including user code), or even the total memory that is used by spark.  "spark.shuffle.memoryFraction" has been around practically forever, so I doubt existing users will see the connection to this metric, but then again I suppose we can't be too hamstrung by legacy.

I also wonder if it just be named something far more specific to SQL + Tungsten if that is the only case its accurately measured.